vtex/affiliates
# Feature/change verify user affiliate lead from event to route

#### What is the purpose of this pull request?

- Update verifyUserAffiliateLead handler to be triggered via an REST API request, instead of using a courier event.
- Add the new app [vtex.affiliates-order-form-spy](https://github.com/vtex-apps/affiliates-order-form-spy) as a dependency.

#### What problem is this solving?

The app was using an event sent by [vtex.order-form-courier](https://github.com/vtex-apps/order-form-courier) to add the affiliateId on customData when a user had an AffiliateLead. This operation is critical and must be handled using a service call instead of being inserted in an event queue, to ensure it is being consistently completed on time during checkout.

To this end, a new [app](https://github.com/vtex-apps/affiliates-order-form-spy) was created and used as a dependency to listen to Order Form changes during checkout and notify the service to update customData accordingly.

#### How to test it?

Complete a purchase using an valid [affiliate store](https://gabiru--sandboxbrdev.myvtex.com/affiliates/fulano) to be marked as an Lead for that affiliate, then complete a purchase using the [main store](https://gabiru--sandboxbrdev.myvtex.com). Verify that the notification is sent during the second checkout and that the customData on you orderForm is updated accordingly.

#### How does this PR make you feel? [:link:](http://giphy.com/)



![](https://media.giphy.com/media/Oj5w7lOaR5ieNpuBhn/giphy.gif)

#### Types of changes
